source/mod_units.f90

```diff
@@ -483,6 +483,100 @@ subroutine f_flush(unit)
 
 end subroutine f_flush
 
+! =================================================================================================
+!  Reposition Text File
+!
+!  V.K. Berglyd Olsen, BE-ABP-HSS
+!  Created: 2019-07-24
+!  Updated: 2019-12-19
+!
+!  The routine was moved from Scatter Module, and was originally intended for Checkpoint/Restart,
+!  but should also work outside of C/R.
+!
+!  Parameters:
+!   * fileName : The file name of the file to be opened. Needed both to ensure not multiple units
+!                is assigned, and to make log messages more readable.
+!   * fileUnit : The file unit variable for the file. The unit can be assigned already, but must
+!                not be open. This routine opens the file, and returns the unit.
+!   * filePos  : The file position variable used to track the latest position in the file. This is
+!                the variable the calling code must keep track on and checkpoint.
+!   * newPos   : The desired position the file should be truncated to. This is the value the
+!                calling code should restore from checkpoint file. It is the checkpointed value of
+!                filePos.
+! =================================================================================================
+subroutine f_positionFile(fileName, fileUnit, filePos, newPos)
+
+  use parpro
+  use crcoall
+
+  character(len=*), intent(in)    :: fileName
+  integer,          intent(inout) :: fileUnit
+  integer,          intent(inout) :: filePos
+  integer,          intent(in)    :: newPos
+
+  logical isOpen, fErr
+  integer iError
+  integer j
+  character(len=mInputLn) aRecord
+
+  call f_requestUnit(fileName,fileUnit)
```

Comment:
shouldn't this call be under if statement? `f_requestUnit` initialises fileUnit to -1

Reply:
What should the if-statement check?

Reply:
clarified by discussion - fine